src/MaestroConfiguration/src/Microsoft.DotNet.MaestroConfiguration.Client/ConfigurationRepositoryManager.cs

```diff
@@ -189,6 +260,93 @@ private async Task CreatePullRequest(
+
+        // If not in the default file, search all files in the folder for that yaml type
+        _logger.LogInformation("Couldn't file configuration object at the default location. Searching all files in folder... this might take a few minutes");
+        return await SearchAllFilesInFolderAsync(gitRepo, repositoryUri, workingBranch, searchObject);
+    }
+
+    private async Task<(string FilePath, List<T> Content)?> TryFindInDefaultFileAsync<T>(
+        IGitRepo gitRepo,
+        string repositoryUri,
+        string workingBranch,
+        T searchObject)
+        where T : IYamlModel
+    {
+        var defaultFilePath = ConfigFilePathResolver.GetDefaultFilePath(searchObject);
+
+        try
+        {
+            var defaultFileContents = await gitRepo.GetFileContentsAsync(repositoryUri, workingBranch, defaultFilePath);
+            var deserializedYamls = _yamlDeserializer.Deserialize<List<T>>(defaultFileContents);
+
+            if (ContainsYamlModel(deserializedYamls, searchObject))
+            {
+                return (defaultFilePath, deserializedYamls);
+            }
+        }
+        catch (FileNotFoundInRepoException)
+        {
+            // Default file doesn't exist
+        }
+
+        return null;
+    }
+
+    private async Task<(string FilePath, List<T> Content)> SearchAllFilesInFolderAsync<T>(
+        IGitRepo gitRepo,
+        string repositoryUri,
+        string workingBranch,
+        T searchObject)
+        where T : IYamlModel
+    {
+        var folderPath = ConfigFilePathResolver.GetDefaultFileFolder(searchObject);
+        var searchKey = YamlModelUniqueKeys.GetUniqueKey(searchObject);
+
+        // Get list of all files in the folder
+        var filePaths = await gitRepo.ListBlobsAsync(repositoryUri, workingBranch, folderPath);
+
+        // Search each file one by one for the object
+        foreach (var filePath in filePaths)
+        {
+            try
+            {
+                var fileContent = await gitRepo.GetFileContentsAsync(repositoryUri, workingBranch, filePath);
+                var deserializedYamls = _yamlDeserializer.Deserialize<List<T>>(fileContent);
+
+                if (ContainsYamlModel(deserializedYamls, searchObject))
+                {
+                    _logger.LogInformation("Search string {0} found in {1}", searchKey, filePath);
+                    return (filePath, deserializedYamls);
+                }
+            }
+            catch (FileNotFoundInRepoException)
+            {
+                // File was listed but couldn't be read, skip it
+                continue;
+            }
+        }
+
+        throw new ArgumentException($"No object with id {searchKey} was found on branch {workingBranch}");
+    }
+
+    private static bool ContainsYamlModel<T>(IReadOnlyCollection<T> yamlModels, T searchObject)
+        where T : IYamlModel
+        => yamlModels.Any(y => YamlModelUniqueKeys.GetUniqueKey(y) == YamlModelUniqueKeys.GetUniqueKey(searchObject));
```

Comment:
We should not be computing `YamlModelUniqueKeys.GetUniqueKey(searchObject)` over and over again, we should pull it out of the Linq